src/Kiota.Builder/Manifest/RequestInfoComparer.cs

```diff
@@ -8,13 +8,18 @@ public class RequestInfoComparer : IEqualityComparer<RequestInfo>
 {
     public bool Equals(RequestInfo? x, RequestInfo? y)
     {
-        return x == null && y == null || x != null && y != null && GetHashCode(x) == GetHashCode(y);
+        if (x is null || y is null) return object.Equals(x, y);
+        const StringComparison comparison = StringComparison.OrdinalIgnoreCase;
+        return string.Equals(x.Method, y.Method, comparison) && string.Equals(x.UriTemplate, y.UriTemplate, comparison);
     }
 
     public int GetHashCode([DisallowNull] RequestInfo obj)
     {
-        if (obj == null) return 0;
-        return (string.IsNullOrEmpty(obj.Method) ? 0 : obj.Method.GetHashCode(StringComparison.OrdinalIgnoreCase)) * 7 +
-            (string.IsNullOrEmpty(obj.UriTemplate) ? 0 : obj.UriTemplate.GetHashCode(StringComparison.OrdinalIgnoreCase)) * 3;
+        var hash = new HashCode();
+        if (obj == null) return hash.ToHashCode();
+        var comparer = StringComparer.OrdinalIgnoreCase;
```

Comment:
you could assign the comparer to a read only static field instead and use the comparer both here and in the equals method

Reply:
Also, the type used in the Equals method is different from the one used in GetHashCode. i.e.

String***Comparison***.OrdinalIgnoreCase vs String***Comparer***.OrdinalIgnoreCase

Reply:
which shouldn't be a problem if we replace the call to `string.Equals` by `stringComparerInstance.Equals`, no?

As for the instance field, do you mean so it can be passed as a constructor parameter? I don't have a strong opinion here even though we don't have an immediate need for it.